### DISCUSSION TITLE
Selectively enable / disable actions

### DISCUSSION THREAD

**Discussion by dsathyakumar** [statelyai/xstate#2117] [Q&A] [answered]
Hello 👋 

Are there means to selectively enable some actions ?
I'm working on an uploader with a spawned background logger service that flushes logs finally upon reaching a `diagnostics` state.
All the while, logs are sent to the service via `send` actions, that are wired up at various event transitions & state entry exit points.
```
entry: ["logUploadEntry"]
```
The logger accumulates these logs & finally upon reaching the `diagnostiscs` state, they are flushed. 

Just wondering, if there was a way to selectively enable / disable logging actions based on a config.

**Comment by VanTanev** [comment]
You could do something like this to use the `NODE_ENV`;

```typescript
actions: {
  logUploadEntry: (context, event) => {
    if (process.env.NODE_ENV === development) {
      // your normal logging code
    }
  }
}
```

Or something like this if you put the config on `context`: `{ loggingEnabld: boolean }`
```typescript
actions: {
  logUploadEntry: (context, event) => {
    if (context.loggingEnabled) {
      // your normal logging code
    }
  }
}
```

**Comment by davidkpiano** [comment, marked as answer]
The `choose()` action creator also helps here:

```js
entry: choose([
  { actions: ['logUploadEntry'], cond: 'loggingEnabled' }
]);
```

https://xstate.js.org/docs/guides/actions.html#choose-action
